recipes/LibriSpeech/ASR/transducer/train.py

```diff
@@ -69,10 +68,62 @@ def compute_forward(self, batch, stage):
             )
 
         current_epoch = self.hparams.epoch_counter.current
+
+        # Default to infinite context visibility
+        transformer_chunk_size = -1
+        left_context_chunks = -1
+
+        # Old models may not have the streaming hparam, we don't break them in
+        # any other way so just check for its presence
+        if hasattr(self.hparams, "streaming") and self.hparams.streaming:
+            # TODO: while this is fairly small logic, it may make sense to
+            # extract it to its own class orchestrating dynamic chunk training,
+            # partly because explantions are beneficial
+            if stage == sb.Stage.TRAIN:
+                # When training for streaming, for each batch, we have a
+                # `dynamic_chunk_prob` probability of sampling a chunk size
+                # between `dynamic_chunk_min` and `_max`, otherwise output
+                # frames can see anywhere in the future.
+                # NOTE: We use torch random to be bound to the experiment seed.
+                if torch.rand((1,)).item() < self.hparams.dynamic_chunk_prob:
```

Comment:
Couldn't we put this on the audio_pipeline function to generate these masks on CPU and benefit from parallel workers?

Reply:
In theory, yes, in practice, it's currently only <5% of the runtime in my testing without even computing the loss or the backward, so idk if it is worth the micro-optimization (as it won't affect streaming where we don't need to mask).